phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java

```diff
@@ -129,45 +128,42 @@ public void testIndexRebuildTask() throws Throwable {
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
                     TENANT1, null, viewName,
                     PTable.TaskStatus.CREATED.toString(), data, null, startTs, null, true);
-
-
+            conn.commit();
             task.run();
+            waitForTaskState(conn, PTable.TaskType.INDEX_REBUILD, viewName, startTs, PTable.TaskStatus.COMPLETED);
 
             String viewIndexTableName = MetaDataUtil.getViewIndexPhysicalName(baseTable);
             ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
             int count = getUtility().countRows(queryServices.getTable(Bytes.toBytes(viewIndexTableName)));
             assertTrue(count == numOfValues);
 
-
             // Remove index contents and try again
             Admin admin = queryServices.getAdmin();
             TableName tableName = TableName.valueOf(viewIndexTableName);
             admin.disableTable(tableName);
-            admin.truncateTable(tableName, false);
+            admin.truncateTable(tableName, true);
 
             data = "{IndexName:" + indexName + ", DisableBefore:true}";
 
-            // Add a new task (update status to created) to System.Task to rebuild indexes
+            // Update task to to rebuild indexes.
             Task.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
                     TENANT1, null, viewName,
                     PTable.TaskStatus.CREATED.toString(), data, null, startTs, null, true);
+            conn.commit();
```

Comment:
Why do we need `conn.commit` here?

Reply:
@gokceni I still don't understand why we are committing here.

Reply:
removed